src/OnboardingSPA/components/SocialMediaForm/index.js

```diff
@@ -108,7 +108,40 @@ const SocialMediaForm = ({ socialData, setSocialData, setIsValidSocials }) => {
             setIsValidSocials(false);
     }
 
+    const isValidHandle = (handle) => {
+        return handle.match(`^[A-Za-z0-9_]{1,25}$`) ? true : false;
+    }
+
+    const isValidTwitterUrl = (url) => {
+        return url.match(`^http(?:s)?:\/\/(?:www\.)?twitter\.com\/([A-Za-z0-9_]{1,25})\/?$`) ? true : false;
+    }
+
+    const checkValidTwitterUrl = function(socialInput, data) {
+        data = data.substring(data.indexOf('@') + 1);
+
+        if( !isValidHandle(data) && !isValidTwitterUrl(data)) { // check for @handle and twitter url
```

Comment:
If we are just checking these two conditions we can just move this line to **checkValidUrl** so that we can almost remove the whole logic which is already there in checkValidUrl, so as to reduce code duplicacy.

Reply:
done